pkg/auth/grpc.go

```diff
@@ -50,9 +52,21 @@ func (ai *AuthInterceptor) validateToken(md metadata.MD) (*AuthInfo, bool) {
 	}
 
 	tokenKey := strings.TrimPrefix(md["authorization"][0], "Bearer ")
-	token, workspace, err := ai.backendRepo.AuthorizeToken(context.TODO(), tokenKey)
-	if err != nil {
-		return nil, false
+
+	var token *types.Token
+	var workspace *types.Workspace
+
+	token, workspace, err := ai.workspaceRepo.AuthorizeToken(tokenKey)
+	if err != nil || token == nil || workspace == nil {
```

Comment:
I think it would be simpler to just return an error from AuthorizeToken even if the contents of the key is "".

Whats the benefit of checking each of these?

Reply:
done